test/e2e/affinity_group.go

```diff
@@ -79,6 +79,17 @@ func AffinityGroupSpec(ctx context.Context, inputGetter func() CommonSpecInput)
 }
 
 func executeTest(ctx context.Context, input CommonSpecInput, namespace *corev1.Namespace, specName string, clusterResources *clusterctl.ApplyClusterTemplateAndWaitResult, affinityType string) []string {
+	csClient := CreateCloudStackClient(ctx, input.BootstrapClusterProxy.GetKubeconfigPath())
+	zoneName := input.E2EConfig.GetVariable("CLOUDSTACK_ZONE_NAME")
+	numHosts := GetHostCount(csClient, zoneName)
+	machineCount := int64(3) // Must allocate an odd number of CP machines w/ stacked etcd
+	if numHosts < 3 {
+		// If not enough ACS hosts to support 3 machines with anti-affinity, fall back to just allocating one machine.
+		//  Test will only confirm that the single VM is in an anti-affinity group, but not that ACS anti-affinty works.
+		By("Fewer than three host in ACS env.  Only verifying that a single VM gets placed in an AG, not that true affinity/anti-affinity is ultimately achieved.")
```

Comment:
I think we should explicitly skip this test if there are <3 hosts. This way is very subtle and tests some plumbing code but not the full vertical feature. If we skip the test with a message explaining why, then the user knows they should address it in their environment.

Reply:
Done. That makes the most sense. Thanks for that Max.